src/index/mod.rs

```diff
@@ -42,14 +42,15 @@ impl Index {
         let url = repository_url.clone();
         let diff = crates_index_diff::Index::from_path_or_cloned_with_options(
             &path,
-            crates_index_diff::CloneOptions {
+            crates_index_diff::index::CloneOptions {
```

Comment:
This seems to fail in our build-tests: 

```
thread 'main' panicked at 'failed to initialize the index: initialising registry index repository  Caused by:     error reading from the zlib stream; class=Zlib (5)
```

Reply:
I can actually reproduce this issue with one of my tools, `criner`, but thus far managed to talk myself into "it's probably something with my local index and nuking it should fix it". Clearly, that is not the case and I am investigating.

It's very strange to see this error as I believe the code for fetching things or cloning didn't change at all, and how can it be that it's entirely within libgit2? Anyway, I shall _hopefully_ find out soon enough :D.

Reply:
I could definitely validate that v10 of `crates-index-diff` works, whereas v11 will start streaming the pack but abort with the above message after a couple of seconds. I have a feeling it's less related to zlib, but more to the connection aborting for some reason. `open-ssl` versions are all the same, I tried the vendored openssl feature, as well as `zlib-ng-compat`, but nothing changed.
Thus, the `diff` should reveal the issue, or a binary search.

Reply:
It turned out to be a misconfigured zlib which caused the issue. Overall, that's more than 2h of my life I am not going to get back, but at least it's a 2 for 1 deal as I believe `cargo` would have exhibited the same issue 🎉.
And, it's an additional motivator to get `gitoxide` into shape for fetching repositories and finish the transition of `crates-index-diff` 😁.